packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -204,11 +206,16 @@ export class TerminalWidget extends BaseWidget implements StatefulWidget {
      * new terminal widget.
      * If id is provided attach to the terminal for this id.
      */
-    async start(id?: number): Promise<void> {
+    async start(id?: number): Promise<number> {
         this.terminalId = typeof id !== 'number' ? await this.createTerminal() : await this.attachTerminal(id);
         this.resizeTerminalProcess();
         this.connectTerminalProcess();
+        if (this.terminalId) {
+            return this.terminalId;
+        }
+        return Promise.reject(undefined);
```

Comment:
Please replace with a proper error: `throw new Error('Failed to start a terminal for the id: ' + id);`.